docs/DifferentiableProgrammingImplementation.md

```diff
@@ -0,0 +1,1012 @@
+    - This seems more efficient because reabstraction thunk JVPs/VJPs only need to call the original function’s JVP/VJP - the other elements of the `@differentiable` function are not relevant.
+    - It may be possible that the derivative of a reabstraction thunk is simply a reabstraction thunk for the derivative function’s type. If so, this would be a significant simplification: within the reabstraction thunk derivative, we can simply call another reabstraction thunk and avoid other code generation (e.g. reabstraction thunk differential/pullback functions).
+    - `JVPEmitter::visitApplyInst` and `VJPEmitter::visitApplyInst` need special case logic to transform the single `apply` in reabstraction thunks into an `apply` of the “JVP/VJP” function-typed argument.
+
+## Future directions and infrastructural changes
+
+Below is a list of anticipated infrastructure changes to the Swift differentiation system. Non-core changes (e.g. API changes) are not listed here.
+
+### Differential-first (forward-mode) automatic differentiation
+
+Differential-first automatic differentiation is currently a work-in-progress. It is not yet at parity with the existing pullback-first (i.e. reverse-mode) automatic differentiation support.
+
+See [here](https://github.com/apple/swift/blob/main/docs/DifferentiableProgramming.md#differential-operators) for more information about forward-mode differential operators in Swift.
+
+### Linear maps and transposition
+
+Linear maps are a fundamental concept in differentiation. Since differentiation is linear approximation, the derivative of a linear map is itself.
+
+We plan to add support for:
+- Linear map function types (`@differentiable(linear)` function types), which are a subtype of `@differentiable` (and infinitely differentiable) function types.
+- Linear map transposition as a first-class operation. This unlocks a correspondence between forward-mode and reverse-mode differentiation: `differentials and pullbacks are transposes of each other`.
+
+See [here](https://github.com/apple/swift/blob/main/docs/DifferentiableProgramming.md#linear-maps) for more information about linear maps. With linear maps and transposition, the differentiation system will change in the following ways:
+- All JVP functions will return a `@differentiable(linear)`-typed differential instead of a normal function-typed differential.
+- VJP functions will be removed throughout the differentiation system.
+- Pullback function generation will change to use transposition.
+
+## How to contribute? 🤔
+
+[JIRA link to AutoDiff starter bugs](https://bugs.swift.org/issues/?jql=project%20%3D%20TF%20AND%20status%20%3D%20Open%20AND%20component%20in%20(Autodiff)%20AND%20labels%20in%20(StarterBug)). As of the time of this writing, there are no open starter bugs. If you are interested in helping with differentiable programming in Swift, please email [swift@tensorflow.org](mailto:swift@tensorflow.org) and we can help you find some starter tasks.
+
```

Comment:
```suggestion
```

Leaving the JIRA link might be okay as well, if they are still relevant (but that deserves a proper scrub).  The mailto is not appropriate, and this work should encourage contributions as per the regular Swift development practices.